plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/sqs.ts

```diff
@@ -121,45 +121,61 @@ export class SqsServiceExtension implements ServiceExtension {
     tracer: Tracer,
     config: AwsSdkInstrumentationConfig
   ) => {
-    const messages: SQS.Message[] = response?.data?.Messages;
-    if (messages) {
-      const queueUrl = this.extractQueueUrl(response.request.commandInput);
-      const queueName = this.extractQueueNameFromUrl(queueUrl);
-
-      pubsubPropagation.patchMessagesArrayToStartProcessSpans<SQS.Message>({
-        messages,
-        parentContext: trace.setSpan(context.active(), span),
-        tracer,
-        messageToSpanDetails: (message: SQS.Message) => ({
-          name: queueName ?? 'unknown',
-          parentContext: propagation.extract(
-            ROOT_CONTEXT,
-            extractPropagationContext(
-              message,
-              config.sqsExtractContextPropagationFromPayload
-            ),
-            contextGetter
-          ),
-          attributes: {
-            [SemanticAttributes.MESSAGING_SYSTEM]: 'aws.sqs',
-            [SemanticAttributes.MESSAGING_DESTINATION]: queueName,
-            [SemanticAttributes.MESSAGING_DESTINATION_KIND]:
-              MessagingDestinationKindValues.QUEUE,
-            [SemanticAttributes.MESSAGING_MESSAGE_ID]: message.MessageId,
-            [SemanticAttributes.MESSAGING_URL]: queueUrl,
-            [SemanticAttributes.MESSAGING_OPERATION]:
-              MessagingOperationValues.PROCESS,
-          },
-        }),
-        processHook: (span: Span, message: SQS.Message) =>
-          config.sqsProcessHook?.(span, { message }),
-      });
-
-      pubsubPropagation.patchArrayForProcessSpans(
-        messages,
-        tracer,
-        context.active()
-      );
+    switch (response.request.commandName) {
+      case 'SendMessage':
+        span.setAttribute(
+          SemanticAttributes.MESSAGING_MESSAGE_ID,
+          response.data.MessageId
```

Comment:
Is it guaranteed that data is not nullish? I recommend using optional chaining, so we don't throw if `data` is not valid.

Reply:
Fixed. It's strange that `any` is used. It would be better to refactor the code to use `unknown` to avoid this kind of issues in the future.